packages/compiler-vapor/src/transforms/vIf.ts

```diff
@@ -52,37 +54,109 @@ export function processIf(
         positive: branch,
       })
     }
+  } else {
+    // check the adjacent v-if
+    const siblings = context.parent!.node.children
+    const comments = []
+    let sibling: TemplateChildNode | undefined
+    let i = siblings.indexOf(node)
+    while (i-- >= -1) {
+      sibling = siblings[i]
+
+      if (sibling && sibling.type === NodeTypes.COMMENT) {
+        context.removeNode(sibling)
+        __DEV__ && comments.unshift(sibling)
+        continue
+      }
+
+      if (
+        sibling &&
+        sibling.type === NodeTypes.TEXT &&
+        !sibling.content.trim().length
+      ) {
+        context.removeNode(sibling)
```

Comment:
Just skip it and don't remove it

Reply:
Not deleting nodes can lead to some other problems. You can reproduce the problem by removing these codes and then running unit tests.